src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -445,7 +445,14 @@ internal partial class BoundCatchBlock : ICatchClause
 
         IOperation ICatchClause.Filter => this.ExceptionFilterOpt;
 
-        ILocalSymbol ICatchClause.ExceptionLocal => this.LocalOpt;
+        ILocalSymbol ICatchClause.ExceptionLocal
+        {
+            get
+            {
+                var local = this.Locals.FirstOrDefault();
+                return local?.DeclarationKind == LocalDeclarationKind.CatchVariable ? local : null;
```

Comment:
After a conversation with @AlekseyTs, he explained that the comment in BoundNodes.xml, while correct, can be clarified to say that *if* there is an exception local, it is always first in this list. He's planning to clarify that in a separate PR.